west: runners: uf2: Add more fstype check for is_uf2_partition.

ihidchaos opened the pull request:
The current check code checks the partition type by hard-coding it. 

During testing it was found that if the partition type was displayed as **msdos** it could not be detected correctly, so this partition type determination was added.

<img width="1142" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/20263334/3c5c941f-c50f-4926-8da3-ea22915fb603">
<img width="1829" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/20263334/4ddf10fb-3680-48cf-b073-e6c20bc01e10">

However, maybe a better approach is to refer to the code at [microsoft/uf2](https://github.com/microsoft/uf2/blob/master/utils/uf2conv.py#L243) to avoid hard-coding and additional partition type checking. 

If you think the code in the link is a better fit, I'm more than happy to continue updating the current detection logic.

ihidchaos commented:
Hi @swinslow @carlescufi @mbolivar-nordic, Do you have any thoughts? 😀

kartben commented:
@ihidchaos could you address the compliance checks? You can refer to the contribution guidelines if you need help there. Thanks!

Commit:
west: runners: uf2: Add more fstype check for is_uf2_partition

During testing it was found that if the partition type was displayed as
msdos it could not be detected correctly, so this partition type
determination was added.

Signed-off-by: Yuchao Guo <yuchao.guo@hoorii.io>

ihidchaos commented:
> @ihidchaos could you address the compliance checks? You can refer to the contribution guidelines if you need help there. Thanks!

I updated the signature information and the check now shows "Run compliance checks on patch series (PR) Expected - Waiting for status to be reported", Is that right?

ihidchaos commented:
I see the workflow:
```
Run compliance checks on patch series (PR) Expected - Waiting for status to be reported Required
Scan code for licenses Expected - Waiting for status to be reported Required
```
It's been stuck for a long time, do these jobs need to be triggered manually?

ihidchaos commented:
@kartben 
All checks have passed. Can you help merge this request? Thanks a lot.

kartben reviewed (approved):
looks fine for me but this will need other reviewers. I like the idea of potentially leveraging microsoft/uf2 more.

ihidchaos commented:
Hi @mbolivar-nordic ,Can you help merge this PR?😊
